pkg/lib/registry/registry.go

```diff
@@ -128,23 +128,40 @@ func unpackImage(ctx context.Context, reg image.Registry, ref image.Reference) (
 
 func populate(ctx context.Context, loader registry.Load, graphLoader registry.GraphLoader, querier registry.Query, reg image.Registry, refs []image.Reference, mode registry.Mode, overwrite bool) error {
 	unpackedImageMap := make(map[image.Reference]string, 0)
+	expectedBundles := make(map[string]registry.Package)
+	overwrittenBundles := make(map[string][]string)
 	for _, ref := range refs {
 		to, from, cleanup, err := unpackImage(ctx, reg, ref)
 		if err != nil {
 			return err
 		}
 		unpackedImageMap[to] = from
 		defer cleanup()
-	}
 
-	overwriteImageMap := make(map[string]map[image.Reference]string, 0)
-	if overwrite {
-		// find all bundles that are attempting to overwrite
-		for to, from := range unpackedImageMap {
-			img, err := registry.NewImageInput(to, from)
+		img, err := registry.NewImageInput(to, from)
+		if err != nil {
+			return err
+		}
+		if _, ok := expectedBundles[img.Bundle.Package]; !ok {
+			pkg, err := graphLoader.Generate(img.Bundle.Package)
```

Comment:
does the behavior of graphLoader actually match the real loading behavior? I don't think it's actually used for replaces.